WcaOnRails/app/webpacker/components/Panel/Leader/GroupsManager.jsx

```diff
@@ -21,8 +22,43 @@ function GroupTable({ groupId }) {
   const confirm = useConfirm();
   const { save, saving } = useSaveAction();
 
+  const statusObjectOfGroupType = (groupType) => {
+    switch (groupType) {
+      case groupTypes.teams_committees:
+        return teamsCommitteesStatus;
+      case groupTypes.councils:
+        return councilsStatus;
+      default:
+        return null;
+    }
+  };
+
   const isLead = (role) => role.metadata.status === 'leader';
 
+  const canPromote = (role) => (
+    role.metadata.status === statusObjectOfGroupType(role.group.group_type).member
+  );
+
+  const canDemote = (role) => (
+    role.metadata.status === statusObjectOfGroupType(role.group.group_type).senior_member
+  );
```

Comment:
I think these functions could be lifted out of the component.

Reply:
Done.